library/src/main/java/com/otaliastudios/zoom/ZoomEngine.kt

```diff
@@ -735,38 +766,38 @@ internal constructor(context: Context) : ViewTreeObserver.OnGlobalLayoutListener
      *
      * @return the pan correction to be applied to get into a valid state (0 if valid already)
      */
+    @SuppressLint("RtlHardcoded")
     @ScaledPan
     private fun checkPanBounds(horizontal: Boolean, allowOverScroll: Boolean): Float {
         @ScaledPan val value = if (horizontal) scaledPanX else scaledPanY
         val containerSize = if (horizontal) mContainerWidth else mContainerHeight
         @ScaledPan val contentSize = if (horizontal) mContentScaledWidth else mContentScaledHeight
         val overScrollable = if (horizontal) mOverScrollHorizontal else mOverScrollVertical
         @ScaledPan val overScroll = (if (overScrollable && allowOverScroll) maxOverScroll else 0).toFloat()
+        val alignment = if (horizontal) Alignment.getHorizontal(mAlignment) else Alignment.getVertical(mAlignment)
 
         var min: Float
         var max: Float
         if (contentSize <= containerSize) {
-            // If content is smaller than container, act according to the smaller policy.
+            // If content is smaller than container, act according to the alignment.
             // Expect the output to be >= 0, we will show part of the container background.
             val extraSpace = containerSize - contentSize // > 0
-            when (mSmallerPolicy) {
-                ZoomApi.SMALLER_POLICY_FROM_TRANSFORMATION -> {
-                    // Apply the transformation gravity.
-                    val correction = applyGravity(mTransformation, extraSpace, horizontal)
-                    min = correction
-                    max = correction
-                }
-                ZoomApi.SMALLER_POLICY_CENTER  -> {
-                    // Stay centered. Need a positive translation, that shows some background.
-                    min = extraSpace / 2f
-                    max = extraSpace / 2f
-                }
-                ZoomApi.SMALLER_POLICY_NONE -> {
-                    // Everything is fine, just don't exit the container boundaries.
-                    min = 0f
-                    max = extraSpace
-                }
-                else -> throw IllegalStateException("Unsupported policy: $mSmallerPolicy")
+            val correction = when (alignment) {
+                Alignment.TOP -> applyGravity(Gravity.TOP, extraSpace, horizontal)
+                Alignment.BOTTOM -> applyGravity(Gravity.BOTTOM, extraSpace, horizontal)
+                Alignment.LEFT -> applyGravity(Gravity.LEFT, extraSpace, horizontal)
+                Alignment.RIGHT -> applyGravity(Gravity.RIGHT, extraSpace, horizontal)
+                Alignment.CENTER_VERTICAL -> applyGravity(Gravity.CENTER_VERTICAL, extraSpace, horizontal)
+                Alignment.CENTER_HORIZONTAL -> applyGravity(Gravity.CENTER_HORIZONTAL, extraSpace, horizontal)
+                else -> null
+            }
```

Comment:
This is a lot of code duplication, maybe extract the `applyGravity()` part like this?:

```
val gravity = when (alignment) {
    Alignment.TOP -> Gravity.TOP
    Alignment.BOTTOM -> Gravity.BOTTOM
    Alignment.LEFT -> Gravity.LEFT
    Alignment.RIGHT -> Gravity.RIGHT
    Alignment.CENTER_VERTICAL -> Gravity.CENTER_VERTICAL
    Alignment.CENTER_HORIZONTAL -> Gravity.CENTER_HORIZONTAL
    else -> null
}

val correction = if (gravity != null) applyGravity(gravity, extraSpace, horizontal) else null
if (correction != null) {
[...]
```

Still looks not that great though.

Reply:
I think it looks clean now!